dask/dataframe/core.py

```diff
@@ -5946,7 +5924,7 @@ def map_partitions(
             args2.append(arg)
             dependencies.append(arg)
             continue
-        arg = normalize_arg(arg)
+        arg = normalize_arg(arg) if normalize_args else arg
```

Comment:
Could you clarify what things `normalize_arg` is turning into Delayed? Maybe it's worth rethinking `normalize_arg`'s behavior instead. If `normalize_arg` is actually changing what the arguments look like when the function gets invoked at runtime, that's a problem.

Reply:
Yes, sorry for dropping this in without explaining. When a user passes in args or kwargs to `map_partitions`, it uses `normalize_arg` to check if the argument is "large" and should be converted to a delayed object. Right now, the logic used in `normalize_arg` is just a heuristic for `list` objects: If the length is >10, the argument is converted to `Delayed`. This turned out to be a "problem" for a very specific test example, where the aggregation spec was a list of many operators.  However, it is totally fine with me to modify that test to use a shorter list of operators (and leave the `normalize_arg` question for a later day).

Reply:
Note that the test failure was not a functionality error, but a assertion error that there were the same number of HLG layers for two different groupby cases.

Reply:
Got it. So it shouldn't _really_ matter for users (though putting it in a separate key in the graph is maybe a bit inefficient)?

Reply:
Right - It is unlikely to be efficient for groupby aggregations, and it will result in an extra MaterializedLayer for the Delayed object (for now).

Reply:
I decided to remove this change and just modified the "failing" test with a note.